indico_toolkit/association/positioning.py

```diff
@@ -42,6 +42,45 @@ def positioned_above(
             is_above = False
         return is_above
 
+    def positioned_above_overlap(
+            self, above_pos: dict, below_pos: dict, must_be_same_page: bool = True, min_overlap_percent: float = None
+        ) -> bool:
+        """
+        Check if the location of one box is above another and if the lower box's overlap is at least the given percentage.
+        Args:
+            above_pos (dict): the position expected to be above
+            below_pos (dict): the position expected to be below
+            must_be_same_page (bool, optional): required to be on same page. Defaults to True.
+            min_overlap_percent (float, optional): the minimum amount of overlap needed. Defaults to None.
+
+        Returns:
+            bool: True is above_pos is above below_pos and below_pos' amount of overlap is at least min_overlap_percent
+        """
+        if below_pos["page_num"] < above_pos["page_num"]:
+            is_above = False
+        elif below_pos["page_num"] != above_pos["page_num"] and must_be_same_page:
+            is_above = False
+        elif self.xaxis_overlap(above_pos, below_pos):
+            if (
+                below_pos["page_num"] == above_pos["page_num"]
+            ) and not self.yaxis_above(above_pos, below_pos):
+                is_above = False
+            else:
+                is_above = True
+                if not min_overlap_percent:
+                    is_min_overlap = True
+                else:
+                    horizontal_overlap_distance = abs(self.get_horizontal_min_distance(above_pos, below_pos))
+                    horizontal_below_distance = abs(below_pos["bbLeft"] - below_pos["bbRight"])
+                    if horizontal_overlap_distance / horizontal_below_distance < min_overlap_percent:
+                        is_min_overlap = False
+                    else:
+                        is_min_overlap = True
+        else:
+            is_above = False
```

Comment:
It might be cleaner to just set these two variables at the top of the function and then override them when necessary (would probably let you refactor significantly)

Reply:
i.e. set is_above= False and is_min_overlap=True- can probably cut out a lot of your conditionals